Some miscellaneous cleanup in codegen

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47011 ATen DerivedType is dead, long live ATen RegisterDispatchKey
* #47008 Convert from higher order functions to classes in tools.codegen.gen
* #47002 Delete TypeDefault.h and TypeDerived.h codegen entirely.
* #47000 Delete TypeDefault call code generation logic in VariableType
* #46991 Delete SchemaRegister.cpp, make flag operate on TypeDefault.cpp
* #46970 Desugar missing dispatch field into singleton Math entry
* **#46940 Some miscellaneous cleanup in codegen**
* #46938 Delete Vulkan from code generator.

- Remove inaccurate @generated comments
- Delete some dead code
- Delete some unused headers
- Delete unnecessary SparseTypeDerived.cpp template

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D24573971](https://our.internmc.facebook.com/intern/diff/D24573971)

ezyang committed:
Some miscellaneous cleanup in codegen

- Remove inaccurate @generated comments
- Delete some dead code
- Delete some unused headers
- Delete unnecessary SparseTypeDerived.cpp template

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

bdhirsh reviewed (commented):
Looks like there are some build failures in CUDAType.h

ezyang committed:
Update on "Some miscellaneous cleanup in codegen"


- Remove inaccurate @generated comments
- Delete some dead code
- Delete some unused headers
- Delete unnecessary SparseTypeDerived.cpp template

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D24573971](https://our.internmc.facebook.com/intern/diff/D24573971)

[ghstack-poisoned]

ezyang committed:
Update on "Some miscellaneous cleanup in codegen"


- Remove inaccurate @generated comments
- Delete some dead code
- Delete some unused headers
- Delete unnecessary SparseTypeDerived.cpp template

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D24573971](https://our.internmc.facebook.com/intern/diff/D24573971)

[ghstack-poisoned]
